File: jour_24/jour24.pas
program puissance;

{ --- Fonction Powers --- }
function powers(x : integer; y : integer) : real;
var 
    produit : real; 
    i : integer;
begin 
    produit := 1; 
    for i := 1 to y do 
    begin 
        produit := produit * x;
    end;
    powers := produit;
end;
var 
    nb1 : integer;
    nb2 : integer;
    results : real;

begin 
    repeat
        write('Entrez la valeur de x (positif) : ');
        readln(nb1);
        write('Entrez la valeur de y (positif ou nul) : ');
        readln(nb2);
        if (nb1 <= 0) or (nb2 < 0) then 
        begin 
            writeln('Erreur : entrez des nombres positifs !');
        end;
    until (nb1 > 0) and (nb2 >= 0); 
    results := powers(nb1, nb2);
    
    writeln('Resultat : ', nb1, ' ^ ', nb2, ' = ', results:0:2);
    writeln('Appuyez sur Entree pour quitter.');
    readln;
end.

File: jour_24/jour24.c
#include <stdio.h>

//fonction puissance 
double powers(int x, int y) {
    double produit = 1.0;
    for (int i = 1; i <= y; i++) {
        produit = produit * x;
    }
    return produit;
}
//Fonction pour vider le buffer 
void viderBuffer() {
    int c;
    while ((c = getchar()) != '\n' && c != EOF);
}

int main() {
    int nb1, nb2;
    int retour_scan;
    double results;

    do {
        printf("Entrez la valeur de x (positif) : ");
        retour_scan = scanf("%d", &nb1);

        // Si l'utilisateur n'a pas tapé un nombre
        if (retour_scan != 1) {
            printf("Erreur : vous devez entrer un nombre entier.\n");
            viderBuffer(); 
            nb1 = -1;      // On force la condition de boucle
            continue;
        }
        printf("Entrez la valeur de y (positif ou nul) : ");
        retour_scan = scanf("%d", &nb2);
        if (retour_scan != 1) {
            printf("Erreur : vous devez entrer un nombre entier.\n");
            viderBuffer();
            nb2 = -1;
            continue;
        }
        // Vérification des valeurs mathématiques
        if (nb1 <= 0 || nb2 < 0) {
            printf("Erreur : x doit etre > 0 et y doit etre >= 0.\n");
        }
    } while (nb1 <= 0 || nb2 < 0);
    results = powers(nb1, nb2);
    printf("\nResultat : %d ^ %d = %.2f\n", nb1, nb2, results);
    return 0;
}